src/coreclr/vm/jitinterface.cpp

```diff
@@ -10859,41 +10846,68 @@ void CEECodeGenInfo::getHelperFtn(CorInfoHelpFunc    ftnNum,               /* IN
                     {
                         // Cache it for future uses to avoid taking the lock again.
                         hlpFinalTierAddrTable[dynamicFtnNum] = finalTierAddr;
-                        if (pNativeEntrypoint != NULL)
-                        {
-                            pNativeEntrypoint->accessType = IAT_VALUE;
-                            pNativeEntrypoint->addr = finalTierAddr;
-                        }
+                        accessType = IAT_VALUE;
+                        targetAddr = finalTierAddr;
                         goto exit;
                     }
                 }
             }
 
             if (IndirectionAllowedForJitHelper(ftnNum))
             {
+#ifdef FEATURE_PORTABLE_ENTRYPOINTS
+                accessType = IAT_VALUE;
+                targetAddr = (LPVOID)helperMD->GetPortableEntryPoint();
+#else // !FEATURE_PORTABLE_ENTRYPOINTS
                 Precode* pPrecode = helperMD->GetPrecode();
                 _ASSERTE(pPrecode->GetType() == PRECODE_FIXUP);
-                if (pNativeEntrypoint != NULL)
-                {
-                    pNativeEntrypoint->accessType = IAT_PVALUE;
-                    pNativeEntrypoint->addr = ((FixupPrecode*)pPrecode)->GetTargetSlot();
-                }
+                accessType = IAT_PVALUE;
+                targetAddr = ((FixupPrecode*)pPrecode)->GetTargetSlot();
+#endif // FEATURE_PORTABLE_ENTRYPOINTS
                 goto exit;
             }
         }
 
-        pfnHelper = LoadDynamicJitHelper((DynamicCorInfoHelpFunc)dynamicFtnNum).pfnHelper;
+        pfnHelper = LoadDynamicJitHelper((DynamicCorInfoHelpFunc)dynamicFtnNum);
     }
 
-    _ASSERTE(pfnHelper != NULL);
+    //
+    // Static helpers
+    //
 
-    if (pNativeEntrypoint != NULL)
+    _ASSERTE(pfnHelper != (TADDR)NULL);
+
+    accessType = IAT_VALUE;
+
+#ifdef FEATURE_PORTABLE_ENTRYPOINTS
+    targetAddr = (LPVOID)hlpFuncEntryPoints[ftnNum];
+    if (targetAddr == NULL)
+#endif // FEATURE_PORTABLE_ENTRYPOINTS
     {
-        pNativeEntrypoint->accessType = IAT_VALUE;
-        pNativeEntrypoint->addr = (LPVOID)GetEEFuncEntryPoint(pfnHelper);
+        TADDR entryPoint = GetEEFuncEntryPoint(pfnHelper);
+
+#ifdef FEATURE_PORTABLE_ENTRYPOINTS
+        NewHolder<PortableEntryPoint> portableEntryPoint = new PortableEntryPoint();
```

Comment:
```suggestion
        PortableEntryPoint* portableEntryPoint = (void*)(SystemDomain::GetGlobalLoaderAllocator()->GetHighFrequencyHeap()->AllocMem(sizeof(PortableEntryPoint));
```
Loader allocator may make more sense for this.

Reply:
Backout for that is harder. That is why I went with `new`, where the backout logic is far simpler.

Reply:
You can use `AllocMemHolder` that is about as hard to backout as NewHolder.